app/Http/Controllers/CustomerController.php

```diff
@@ -124,14 +129,71 @@ public function update(Request $request, $id)
      */
     public function destroy( $id)
     {
-        
+        try{
         $customers = Customers::find($id);
         $customers->delete();
-        
+        }
+        catch(\Exception $e){
+            return redirect()->back()->with('danger', "You are not authorized to delete this customer.");
+        }
         return redirect()->route('customers.customers.index')->with('success', "Successfully deleted customer");
 
     }
+    // Mail
+    // public function mailCustomerStatements()
+    // {
+    //     $accounting_system_id = $this->request->session()->get('accounting_system_id');
+    //     $customers = Customers::where('accounting_system_id', $accounting_system_id)
+    //     ->whereHas('receiptReference', function($query) {
+    //         $query->Where('status', 'unpaid')
+    //         ->orWhere('status', 'partially_paid')
+    //         ->where('type', 'receipt');
+    //     })->get();
+
+    //     if($customers->isEmpty())
+    //         return redirect()->back()->with('danger', "No pending statements found.");
+    //     foreach($customers as $customer) {
+    //         $data = $customer->toArray();
+    //         // add receipt reference in data
+    //         $receipts = $customer->receiptReference->toArray();
+    //         $receipts += $customer->receiptReference->receipt->toArray();
+
+    //         Mail::to($customer->email)->queue(new MailCustomerStatement ($data, $receipts));
+    //     }
+
+    //     return redirect()->back()->with('success', "Successfully sent customer statements.");     
+    // }
+
+    // Specific customer mail
+    public function mailCustomerStatement($id)
+    {
+        $accounting_system_id = $this->request->session()->get('accounting_system_id');
+        $customers = Customers::where('accounting_system_id', $accounting_system_id)
+        ->whereHas('receiptReference', function($query) {
+            $query->where('type', 'receipt')
+            ->orWhere('status', 'unpaid')
+            ->orWhere('status', 'partially_paid');     
```

Comment:
This may show `unpaid proformas`. I think nesting line 174 and 175 should do. Give this code a try.
```php
$query->where('type', 'receipt')
  ->where(function ($query) {
    $query->where('status', 'unpaid')
      ->orWhere('status', 'partially_paid');
  });
```